### PR TITLE
Fixes a potential failure in clockcult setup, reworks stargazer enchantments

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Stargazer enchantments are now based on components rather than picking a number between 1 and 10.
Stargazer enchantments will no longer be negative.
Fixes a clockcult post_setup issue where clients could disconnect while the map was loading causing their mind to have a null current mob.

**Retained enchantements:**
Sharpness - Item will have extra force added to it
Antimagic - Item will protect the user from magic
Tiny - Item will be tiny and fit in pockets

**New enchantements:**
Burn - Ignites targets, deals burn damage instead
Blocking - Increases blocking of the item
Electricution - Targets get a non-stunning shock on hit.
Penetration - Target will have armour penetration
Soul Tap - Item will heal the user when they attack a concious mob

## Why It's Good For The Game

Fixes an error, makes the enchantments of clockcult slightly less useless.

## Changelog
:cl:
fix: Fixes a bug in clockcult where a client disconnecting right before the round starts could cause the ark to never open
refactor: Refactors stargazer enchantments, they will no longer be negative and are based on components.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Tested all enchantments and they seem to be working

**Comment by BrumalLake** [comment]
burn may be a little annoying to the user if the target just spreads it

**Comment by PowerfulBacon** [comment]
> burn may be a little annoying to the user if the target just spreads it

It goes both ways and weapons can be easilly replaced if the user finds them annoying so I think its fine personally

File: code/game/gamemodes/clock_cult/clockcult.dm [hunk 1]
```diff
@@ -72,6 +72,12 @@ GLOBAL_VAR(clockcult_eminence)
 	main_cult.setup_objectives()
 	//Create team
 	for(var/datum/mind/servant_mind in selected_servants)
+		//Somehow the mind has no mob, ignore them so it doesn't break everything
+		if(!(servant_mind?.current))
```
Comment (Archanial) [inline code/game/gamemodes/clock_cult/clockcult.dm:76]:
```suggestion
		if(!servant_mind?.current)
```

File: code/modules/antagonists/clock_cult/enchantments/_enchantment.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	level = rand(1, max_level)
+	//Apply effect
+	apply_effect(I)
+	//Add in examine effect
+	RegisterSignal(I, COMSIG_PARENT_EXAMINE, .proc/examine)
+
+/datum/component/enchantment/proc/apply_effect(obj/item/target)
+	return
+
+/datum/component/enchantment/proc/examine(datum/source, mob/user, list/examine_list)
```
Comment (Archanial) [inline code/modules/antagonists/clock_cult/enchantments/_enchantment.dm]:
```suggestion
/datum/component/enchantment/proc/on_examine(datum/source, mob/user, list/examine_list)
```

File: code/modules/antagonists/clock_cult/enchantments/_enchantment.dm [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	SIGNAL_HANDLER
+
+	if(examine_description)
```
Comment (Archanial) [inline code/modules/antagonists/clock_cult/enchantments/_enchantment.dm]: early return here

File: code/modules/antagonists/clock_cult/items/clockwork_item.dm [hunk 4]
```diff
@@ -14,3 +14,5 @@
 		. += clockwork_desc
 	else if(desc)
 		. += desc
+
+	SEND_SIGNAL(src, COMSIG_PARENT_EXAMINE, user, .)
```
Comment (Archanial) [inline code/modules/antagonists/clock_cult/items/clockwork_item.dm:18]: What? Why there?

Comment (Archanial) [reply code/modules/antagonists/clock_cult/items/clockwork_item.dm:18]: just call parent

Comment (KubeRoot) [reply code/modules/antagonists/clock_cult/items/clockwork_item.dm:18]: The whole point is not calling parent to obscure the real description of the item.

File: code/modules/antagonists/clock_cult/enchantments/_enchantment.dm [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		return COMPONENT_INCOMPATIBLE
+	//Get random level
+	level = rand(1, max_level)
+	//Apply effect
+	apply_effect(I)
+	//Add in examine effect
+	RegisterSignal(I, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/enchantment/Destroy()
+	UnregisterSignal(I, COMSIG_PARENT_EXAMINE)
```
Comment (PowerfulBacon) [inline code/modules/antagonists/clock_cult/enchantments/_enchantment.dm]:
```suggestion
	UnregisterSignal(parent, COMSIG_PARENT_EXAMINE)
```

File: code/modules/antagonists/clock_cult/enchantments/_enchantment.dm [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	//Get random level
+	level = rand(1, max_level)
+	//Apply effect
+	apply_effect(I)
+	//Add in examine effect
+	RegisterSignal(I, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/enchantment/Destroy()
+	UnregisterSignal(parent, COMSIG_PARENT_EXAMINE)
+	. = ..()
```
Comment (Archanial) [inline code/modules/antagonists/clock_cult/enchantments/_enchantment.dm]:
```suggestion
	return ..()
```

**Review by Archanial — APPROVED** [review]

**Pull request merged**